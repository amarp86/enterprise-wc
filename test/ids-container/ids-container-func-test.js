/**
 * @jest-environment jsdom
 */
import IdsContainer from '../../src/ids-container/ids-container';

describe('IdsContainer Component', () => {
  let container;

  beforeEach(async () => {
    container = new IdsContainer();
    document.body.appendChild(container);
  });

  afterEach(async () => {
    document.body.innerHTML = '';
  });

  it('renders with no errors', () => {
    const errors = jest.spyOn(global.console, 'error');
    const elem = new IdsContainer();
    document.body.appendChild(elem);
    elem.remove();
    expect(document.querySelectorAll('ids-container').length).toEqual(1);
    expect(errors).not.toHaveBeenCalled();
  });

  it('renders correctly', () => {
    expect(container.shadowRoot.innerHTML).toMatchSnapshot();
  });

  it('renders correctly for unscrollable', () => {
    container.scrollable = false;
    expect(container.template()).toEqual('<div class="ids-container" part="container"><slot></slot></div>');
  });

  it('can set and reset scrollable', () => {
    expect(container.scrollable).toEqual('true');
    expect(container.container.getAttribute('tabindex')).toEqual('0');
    container.scrollable = false;
    expect(container.scrollable).toEqual('false');
    expect(container.getAttribute('scrollable')).toEqual('false');
    expect(container.container.getAttribute('tabindex')).toBeFalsy();
    container.scrollable = true;
    expect(container.scrollable).toEqual('true');
    expect(container.getAttribute('scrollable')).toEqual('true');
    expect(container.container.getAttribute('tabindex')).toEqual('0');
  });

  it('supports setting mode', () => {
    container.mode = 'dark';
    expect(container.container.getAttribute('mode')).toEqual('dark');
  });

  it('supports setting version', () => {
    container.version = 'classic';
    expect(container.container.getAttribute('version')).toEqual('classic');
  });
});
