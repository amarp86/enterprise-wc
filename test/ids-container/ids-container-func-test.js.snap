// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdsContainer Component renders correctly 1`] = `"<style nonce=\\"0a59a005\\">.ids-container { background-color: transparent; }</style><div class=\\"ids-container\\" part=\\"container\\" tabindex=\\"0\\"><slot></slot></div>"`;
